modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcClientTimeZoneTest.java

```diff
@@ -176,25 +180,55 @@ public void dynamicParamRespectsTimeZone() throws SQLException {
                     // Client time zone is "GMT".
                     {
                         assertEquals(timestamp("1970-01-01T00:00:00"), rs.getTimestamp(1));
-                        // Since client time zone is GMT, timestamp will contain original UTC value from store.
+                        assertEquals(LocalTime.of(0, 0, 0), rs.getTime(1).toLocalTime());
+                        assertEquals(LocalDate.of(1970, 1, 1), rs.getDate(1).toLocalDate());
+
                         assertEquals(timestamp("1969-12-31T23:00:00"), rs.getTimestamp(2));
+                        assertEquals(LocalTime.of(23, 0, 0), rs.getTime(2).toLocalTime());
+                        assertEquals(LocalDate.of(1969, 12, 31), rs.getDate(2).toLocalDate());
                     }
+                }
+            }
+        }
+
+        TimeZone.setDefault(TimeZone.getTimeZone("GMT+1"));
+
+        try (Connection conn = DriverManager.getConnection(URL)) {
+            try (Statement stmt = conn.createStatement()) {
+                try (ResultSet rs = stmt.executeQuery("SELECT ts, ts_tz FROM test where id=1")) {
+                    assertTrue(rs.next());
 
                     // Session and client time zone are same ("GMT+1").
                     {
-                        TimeZone.setDefault(TimeZone.getTimeZone("GMT+1"));
-
                         assertEquals(timestamp("1970-01-01T00:00:00"), rs.getTimestamp(1));
+                        assertEquals(LocalTime.of(0, 0, 0), rs.getTime(1).toLocalTime());
+                        assertEquals(LocalDate.of(1970, 1, 1), rs.getDate(1).toLocalDate());
+
                         assertEquals(timestamp("1970-01-01T00:00:00"), rs.getTimestamp(2));
+                        assertEquals(LocalTime.of(0, 0, 0), rs.getTime(2).toLocalTime());
+                        assertEquals(LocalDate.of(1970, 1, 1), rs.getDate(2).toLocalDate());
                     }
+                }
+            }
+        }
+
+        TimeZone.setDefault(TimeZone.getTimeZone("GMT+2"));
+
+        try (Connection conn = DriverManager.getConnection(URL)) {
```

Comment:
let's have at least one test case where time zones is passed through connection string